### PR TITLE
Move Extension tests from ieFuncTests to cpuFuncTests

### PR DESCRIPTION
**Pull request by mateusztabaka**

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/extension/extension.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <extension/extension.hpp>
+
+
+TEST_P(OnnxModelWithCustomAbs, Extension) {
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/extension/extension.cpp:8]: `TEST_P` bodies should be in `plugin/shared/src/extension/extension.cpp `

**Review by iefode — CHANGES_REQUESTED** [review]

**Comment by iefode** [comment]
moreover, can we run extension tests for other plugins? If no, please move it to the cpu specific tests

**Comment by mateusztabaka** [comment]
> moreover, can we run extension tests for other plugins? If no, please move it to the cpu specific tests

I moved those to cpu only tests.

File: inference-engine/tests/functional/plugin/cpu/shared_tests_instances/extension/extension.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (C) 2020 Intel Corporation
+// Copyright (C) 2020-2021 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/shared_tests_instances/extension/extension.cpp:1]: Why `shared_test_instances`??? I suppose to move this folder upder `functional/plugin/cpu` folder

File: inference-engine/tests/functional/plugin/cpu/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright (C) 2019-2020 Intel Corporation
+# Copyright (C) 2019-2021 Intel Corporation
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/cpu/CMakeLists.txt:1]: @dmitry-gorokhov Are you OK with these changes?

**Review by dmitry-gorokhov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Pull request merged**